utils/tests_fetcher.py

```diff
@@ -0,0 +1,387 @@
+                # In case of renames, we'll look at the tests using both the old and new name.
+                if diff_obj.a_path != diff_obj.b_path:
+                    code_diff.extend([diff_obj.a_path, diff_obj.b_path])
+                else:
+                    # Otherwise, we check modifications are in code and not docstrings.
+                    if diff_is_docstring_only(repo, commit, diff_obj.b_path):
+                        print(f"Ignoring diff in {diff_obj.b_path} as it only concerns docstrings or comments.")
+                    else:
+                        code_diff.append(diff_obj.a_path)
+
+    return code_diff
+
+
+def get_module_dependencies(module_fname):
+    """
+    Get the dependencies of a module.
+    """
+    with open(module_fname, "r", encoding="utf-8") as f:
+        content = f.read()
+
+    imports = re.findall(r"from\s+(\.+\S+)\s+import\s+\S+\s", content)
+    module_parts = module_fname.split(os.path.sep)
+    dependencies = []
+    for imp in imports:
+        level = 0
+        while imp.startswith("."):
+            imp = imp[1:]
+            level += 1
+
+        if len(imp) > 0:
+            dep_parts = module_parts[: len(module_parts) - level] + imp.split(".")
+        else:
+            dep_parts = module_parts[: len(module_parts) - level] + ["__init__.py"]
+        imported_module = os.path.sep.join(dep_parts)
+        if imported_module.endswith("transformers/__init__.py"):
+            continue
+        if os.path.isfile(f"{imported_module}.py"):
+            dependencies.append(f"{imported_module}.py")
+        elif os.path.isdir(imported_module) and os.path.isfile(os.path.sep.join([imported_module, "__init__.py"])):
+            dependencies.append(os.path.sep.join([imported_module, "__init__.py"]))
+    return dependencies
+
+
+def create_reverse_dependency_map():
+    """
+    Create the dependency map from module filename to the list of modules that depend on it.
+    """
+    modules = [
+        str(f.relative_to(PATH_TO_TRANFORMERS))
+        for f in (Path(PATH_TO_TRANFORMERS) / "src/transformers").glob("**/*.py")
+    ]
+    direct_deps = {m: get_module_dependencies(m) for m in modules}
+
+    something_changed = True
+    while something_changed:
+        something_changed = False
+        for m in modules:
+            for d in direct_deps[m]:
+                for dep in direct_deps[d]:
+                    if dep not in direct_deps[m]:
+                        direct_deps[m].append(dep)
+                        something_changed = True
+
+    reverse_map = collections.defaultdict(list)
+    for m in modules:
+        for d in direct_deps[m]:
+            reverse_map[d].append(m)
+
+    return reverse_map
+
+
+# Any module file that has a test name which is not obviously inferred from its name should go there (better name
+# your test file appropriately than add a new entry in this list if you can!)
```

Comment:
```suggestion
# Any module file that has a test name which can't be inferred automatically from its name should go here. A better 
# approach is to (re-)name the test file accordingly, and second best to add the correspondence map here.
```

Reply:
but what would happen when a test is added which can't be inferred from and its not in this map?